core/block.go

```diff
@@ -330,19 +338,21 @@ func (b *Block) process() Interrupt {
 		b.routing.Source,
 		b.routing.InterruptChan)
 
-	if interrupt != nil {
-		if b.sourceType != NONE && b.sourceType != SERVER {
-			b.routing.Source.Unlock()
+	// unlock the store if necessary
+	if s != nil {
```

Comment:
same down here, don't see the value in keeping `s`? especially when `s` is an interface?